pegasus/src/database.rb

```diff
@@ -63,14 +63,28 @@ def find_with_grade_level(gradelevel)
 
   # return the first tutorial with a matching code
   def find_with_code(code)
-    by_code = CDO.cache.fetch("Tutorials/#{@table}/by_code") {@contents.index_by {|row| row[:code]}}
-    by_code[code]
+    ids_by_code = CDO.cache.fetch("CdoTutorials/#{@table}/ids_by_code") do
+      @contents.index_by {|row| row[:code]}.transform_values {|obj| obj[:id]}
+    end
+    id = ids_by_code[code]
+    if id
+      DB[@table].select(*@column_aliases).where(id: id).first
+    else
+      nil
+    end
```

Comment:
nit; I believe the preferred Ruby style is to use a guard clause rather than a conditional for this pattern. Something like: 

```suggestion
    return nil unless id
    DB[@table].select(*@column_aliases).where(id: id).first
```